src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -738,13 +783,17 @@ EmbeddedAssemblies::typemap_java_to_managed (MonoString *java_type)
 		return nullptr;
 	}
 
-	c_unique_ptr<char> java_type_name {mono_string_to_utf8 (java_type)};
-	if (XA_UNLIKELY (!java_type_name || *java_type_name == '\0')) {
-		log_warn (LOG_ASSEMBLY, "typemap: empty Java type name passed to 'typemap_java_to_managed'");
+	// We need to generate hash for all the bytes, and since MonoString is Unicode, we double the length to get the
+	// number of bytes.
+	int name_len = mono_string_length (java_type) << 1;
```

Comment:
`*2` is clearer than `<<1`. :-)

Reply:
I know, but I've seen that some compilers (admittedly not the ones in the current NDK) produce a multiplication assembler instruction for `*` (and division for `/`) instead of a left shift (right shift), so I chose to use `<<` (and `>>` elsewhere) just to be safe

Reply:
You're kidding.  It's 2022, and compilers don't optimize that?!